nova_vm/src/ecmascript/abstract_operations/type_conversion.rs

```diff
@@ -471,3 +471,87 @@ pub(crate) fn to_object(_agent: &mut Agent, argument: Value) -> JsResult<Object>
         _ => Ok(Object::try_from(argument).unwrap()),
     }
 }
+
+/// ### [7.1.19 ToPropertyKey ( argument )](https://tc39.es/ecma262/#sec-topropertykey)
+pub(crate) fn to_property_key(agent: &mut Agent, argument: Value) -> JsResult<Value> {
+    // 1. Let key be ? ToPrimitive(argument, hint String).
+    let key = to_primitive(agent, argument, Some(PreferredType::String))?;
+
+    // 2. If Type(key) is Symbol, then
+    if key.is_symbol() {
+        // a. Return key.
+        return Ok(key);
+    }
+
+    // 3. Return ! ToString(key).
+    Ok(to_string(agent, key).unwrap().into())
+}
+
+/// ### [7.1.20 ToLength ( argument )](https://tc39.es/ecma262/#sec-tolength)
+pub(crate) fn to_length(agent: &mut Agent, argument: Value) -> JsResult<Value> {
+    // 1. Let len be ? ToIntegerOrInfinity(argument).
+    let len = to_integer_or_infinity(agent, argument)?;
+
+    // 2. If len ≤ 0, return +0𝔽.
+    match len {
+        Number::Integer(n) if n.into_i64() <= 0 => {
+            return Ok(0.0.into());
```

Comment:
**nitpick:** I presume you can combine all the three matches into a single branch with `|`. Or maybe it doesn't work with this syntax?

We could maybe setup a `Default` impl for `Number` that would just gen a zero. Since it's a valid i53 it'll become an `Integer` variant though, so the `0.0` here may be a bit of a red herring to someone reading the code.